src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -760,6 +745,49 @@ private void CheckStake(Block block, ChainedHeader chainTip)
             return true;
         }
 
+        internal Transaction PrepareCoinStakeTransactions(
+            int currentChainHeight,
+            CoinstakeContext coinstakeContext,
+            long coinstakeInputValue,
+            int utxosCount,
+            long amountStaked)
+        {
+            // Split stake into SplitFactor utxos if above threshold.
+            const int SplitFactor = 8;
+            bool shouldSplitStake = this.ShouldSplitStake(
+                utxosCount,
+                amountStaked,
+                coinstakeInputValue,
+                SplitFactor,
+                currentChainHeight);
+
+            if (!shouldSplitStake)
+            {
+                coinstakeContext.CoinstakeTx.Outputs[1].Value = coinstakeInputValue;
+                this.logger.LogTrace("Coinstake output value is {0}.", coinstakeContext.CoinstakeTx.Outputs[1].Value);
+                return coinstakeContext.CoinstakeTx;
+            }
+
+            this.logger.LogTrace("Coinstake UTXO of value {0} will be split into {1}.", coinstakeInputValue, SplitFactor);
+            long splitValue = coinstakeInputValue / SplitFactor;
+            long remainder = coinstakeInputValue - (SplitFactor - 1) * splitValue;
+            coinstakeContext.CoinstakeTx.Outputs[1].Value = remainder;
+
+            for (int i = 1; i < SplitFactor; i++)
```

Comment:
I would probably slightly prefer going from 0 to split factor -1 because this looks like an error for the first look